theano/tensor/nnet/ctc.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import theano
+import theano.tensor as T
+from theano import gof
+from theano.gof import local_optimizer
+from theano.tensor.opt import register_canonicalize
+from theano.tensor.opt import register_stabilize
+from theano.gradient import grad_undefined
+
+import os
+
+class ConnectionistTemporalClassification(gof.COp):
+    __props__ = ()
+
+    func_file = "./ctc_wrapper.c"
+    func_name = "APPLY_SPECIFIC(ctc_cost_cpu)"
+
+    def __init__(self, computeGradient=True):
+        super(ConnectionistTemporalClassification, self).__init__(self.func_file,
+                                                                  self.func_name)
+        self.computeGradient = computeGradient
+        self.costs = T.fvector(name="ctc_cost")
+        if self.computeGradient:
+            self.gradients = T.ftensor3(name="ctc_grad")
+
+        if "CTC_LIB" in os.environ:
```

Comment:
Insted of using environment variables, we generally add a new theano config flag that is used only when it is specified. This makes configuration easier in user side:
 - if the config flag is not specified, g++ will look for headers and libraries in default's OS or Python folders.
 - else Theano will get convenient folders from config flag. This allows user to change library folder quickly if needed.

Here I suggest to add config flag `ctc.root` equivalent to your current `CTC_LIB` environment variable. Check `theano/configdefaults.py` to see how we add a new config variable: https://github.com/Theano/Theano/blob/master/theano/configdefaults.py

As an example, check how we use flags `dnn.include_path` and `dnn.library_path` in new back-end DNN module: `theano/gpuarray/dnn.py`: https://github.com/Theano/Theano/blob/master/theano/gpuarray/dnn.py#L255 .